objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/ObjectRecognitionCore.java

```diff
@@ -390,6 +399,408 @@ public void sendDataFromImage(ImageRetrieverResults imageResults, ExtensionServi
        }
    }
    
+   /**
+    * Uploads requested images to VANTIQ. Called when query parameter "operation" is set to "save".
+    * @param request        The parameters sent with the query.
+    * @param replyAddress   The replyAddress used to send a query response.
+    */
+   public void uploadLocalImages(Map<String, ?> request, String replyAddress) {   
+       String imageDir = null;
+       String imageName = null;
+       String imageDate = null;
+       String dateRange = null;
+       List<String> imageDateList = null;
+       ImageUtil imageUtil;
+       
+       // ------------------------ GATHERING PARATMERS ---------------------------
+       
+       // Checking to make sure there is an image directory specified. (Required)
+       if (request.get("imageDir") instanceof String) {
+           imageDir = (String) request.get("imageDir");
+           
+           // Set up ImageUtil properties
+           imageUtil = setupQueryImageUtil(request);
+           
+           // Checking if "imageName" option was used to specify the file(s) to upload.
+           if (request.get("imageName") instanceof String) {
+               imageName = (String) request.get("imageName");
+               
+           // Checking if "imageDate" option was used to specify the file(s) to upload.
+           } else if (request.get("imageDate") instanceof String) {
+               imageDate = (String) request.get("imageDate");
+               
+               if (request.get("dateRange") instanceof String) {
+                   dateRange = (String) request.get("dateRange");
+               } else {
+                   dateRange = "none";
+               }
+               
+           // Checking if "imageDate" option was used as a list of two dates to specify the files to upload.
+           } else if (request.get("imageDate") instanceof List) {
+               imageDateList = (List<String>) request.get("imageDate");
+               if (imageDateList.size() != 2) {
+                   client.sendQueryError(replyAddress, "io.vantiq.extsrc.objectRecognition.invalidQueryRequest", 
+                           "The imageDate value was a list with more than two elements. Must be a list containing only "
+                           + "[<yourStartDate>, <yourEndDate>].", null);
+                   return;
+               }
+           } else {
+               client.sendQueryError(replyAddress, "io.vantiq.extsrc.objectRecognition.invalidQueryRequest", 
+                       "No imageName or imageDate was specified, or they were incorrectly specified. "
```

Comment:
Another option to dump all these else's with code in them is to have the various error checkers just set set a code & message (& params if required).  Then, one place at the bottom (or wherever we're sure there's no other error), we make a single call to `client.sendQueryError(replyAddr, code, errorText, errorParams)`. Sometimes that makes things a bit clearer.